packages/core-connector-interface/src/indexer.d.ts

```diff
@@ -1,12 +1,25 @@
-import { Balance } from './types'
+import type { Balance, Public, ChannelEntry } from './types'
 
-type Channel = [source: PeerId, destination: PeerId, stake: Balance]
+export type RoutingChannel = [source: PeerId, destination: PeerId, stake: Balance]
+export type ChannelUpdate = { partyA: Public; partyB: Public; channelEntry: ChannelEntry }
 
 declare interface Indexer {
-  getRandomChannel(): Promise<Channel | undefined>
-  getChannelsFromPeer(source: PeerId): Promise<Channel[]>
-  onNewChannels(handler: (newChannels: Channel[]) => void): void
+  start(): Promise<void>
+  stop(): Promise<void>
+
+  getChannelEntry(partyA: Public, partyB: Public): Promise<ChannelEntry | undefined>
+  getChannelEntries(party?: Public, filter?: (node: Public) => boolean): Promise<ChannelUpdate[]>
+
+  // routing
+  getRandomChannel(): Promise<RoutingChannel | undefined>
+  getChannelsFromPeer(source: PeerId): Promise<RoutingChannel[]>
+}
+
+declare interface Indexer {
```

Comment:
Why are we declaring twice?

Reply:
IntelliSense doesn't work otherwise, will add a comment